File: api/src/main/java/com/moebius/api/config/KafkaConfiguration.java
package com.moebius.api.config;

import org.apache.kafka.clients.admin.AdminClient;
import org.apache.kafka.clients.admin.KafkaAdminClient;
import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class KafkaConfiguration {

    @Bean
    public AdminClient kafkaAdminClient(KafkaProperties kafkaProperties){
        var properties = kafkaProperties.buildAdminProperties();
        return KafkaAdminClient.create(properties);
    }
}

File: tracker/src/main/kotlin/com/moebius/tracker/configuration/KafkaConfiguration.kt
package com.moebius.tracker.configuration

import com.moebius.backend.configuration.KafkaConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import

@Configuration
@ComponentScan("com.moebius.backend.service.kafka")
@Import(value = [KafkaConfiguration::class])
class KafkaConfiguration {
}